src/core/Akka.Tests/Pattern/BackoffSupervisorSpec.cs

```diff
@@ -372,46 +388,52 @@ IActorRef WaitForChild()
             var c1 = ExpectMsg<BackoffSupervisor.CurrentChild>().Ref;
             Watch(c1);
             c1.Tell(PoisonPill.Instance);
-            ExpectTerminated(c1);
+            ExpectMsg<Terminated>().ActorRef.Should().Be(c1);
 
-            supervisor.Tell(BackoffSupervisor.GetRestartCount.Instance);
-            ExpectMsg<BackoffSupervisor.RestartCount>().Count.Should().Be(1);
+            AwaitCondition(() =>
+            {
+                supervisor.Tell(BackoffSupervisor.GetRestartCount.Instance);
+                return ExpectMsg<BackoffSupervisor.RestartCount>().Count == 1;
+            });
 
             var c2 = WaitForChild();
-            AwaitAssert(() => c2.ShouldNotBe(c1));
+            c2.ShouldNotBe(c1);
             Watch(c2);
             c2.Tell(PoisonPill.Instance);
-            ExpectTerminated(c2);
+            ExpectMsg<Terminated>().ActorRef.Should().Be(c2);
 
-            supervisor.Tell(BackoffSupervisor.GetRestartCount.Instance);
-            ExpectMsg<BackoffSupervisor.RestartCount>().Count.Should().Be(2);
+            AwaitCondition(() =>
+            {
+                supervisor.Tell(BackoffSupervisor.GetRestartCount.Instance);
+                return ExpectMsg<BackoffSupervisor.RestartCount>().Count == 2;
+            });
 
             var c3 = WaitForChild();
-            AwaitAssert(() => c3.ShouldNotBe(c2));
+            c3.ShouldNotBe(c2);
             Watch(c3);
             c3.Tell(PoisonPill.Instance);
-            ExpectTerminated(c3);
-            ExpectTerminated(supervisor);
+            ExpectMsg<Terminated>().ActorRef.Should().Be(c3);
+            ExpectMsg<Terminated>().ActorRef.Should().Be(supervisor);
         }
 
-        [Fact(Skip = "Racy on Azure DevOps")]
+        [Fact]
         public void BackoffSupervisor_must_stop_restarting_the_child_after_reaching_maxNrOfRetries_limit_using_BackOff_OnFailure()
         {
             EventFilter.Exception<TestException>().Expect(3, () =>
             {
-                var supervisor = Create(OnFailureOptions(maxNrOfRetries: 2));
+                var supervisor = Create(OnFailureOptions(maxNrOfRetries: 2).WithManualReset());
```

Comment:
Supervisor props need to be forced to use manual reset to make sure that it doesn't auto reset the internal restart count mid-spec and fails the spec.